tests/Credentials/GCECredentialsTest.php

```diff
@@ -124,6 +124,37 @@ public function testOnGceWithResidency()
         $this->assertTrue(GCECredentials::onGCE($httpHandler));
     }
 
+    public function testOnWindowsGceWithResidencyWithNoCom()
+    {
+        $method = (new \ReflectionClass(GCECredentials::class))
+            ->getMethod('detectResidencyWindows');
+        
+        $method->setAccessible(true);
+
+        $this->assertFalse($method->invoke(null, 'thisShouldBeFalse'));
+    }
+
+    public function testOnWindowsGceWithResidency()
+    {
+        if (!class_exists(\COM::class)) {
+            throw $this->markTestSkipped('This test only works while running on windiwos with GCE and the COM class enabled');
+        }
+
+        $method = (new \ReflectionClass(GCECredentials::class))
+            ->getMethod('detectResidencyWindows');
+        $method->setAccessible(true);
+
+        $keyPathProperty = (new \ReflectionClass(GCECredentials::class))
+            ->getProperty('REGISTRY_KEY_PATH');
+        $keyPathProperty->setAccessible(true);
+
+        $keyName = (new \ReflectionClass(GCECredentials::class))
+            ->getProperty('REGISTRY_KEY_NAME');
+        $keyName->setAccessible(true);
+
+        $this->assertTrue($method->invoke(null, $keyPathProperty->getValue() . $keyName->getValue()));
```

Comment:
IMHO, to save the use of additional reflection I'd rather just see the hardcoded string.

Also, this test assumes the test is actually running on GCE, which it may not be. I'd suggest simply verifying that the function returns a _bool_, or making it similar to the existing `testOnGceWithResidency`, which is meant to be run manually to test residency directly on the VM. So for instance you could run `vendor/bin/phpunit --filter testOnGceWithResidency` on a Windows GCE VM to test. But it skips when it's not on GCE (which means it doesn't run as part of our Github Actions CI)